src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs

```diff
@@ -0,0 +1,212 @@
+        private ICryptoFunctionService _cryptoFunctionService;
+        private ICryptoService _cryptoService;
+        private IPasswordGenerationService _passwordGenerationService;
+        private IPlatformUtilsService _platformUtilsService;
+        private IEnvironmentService _environmentService;
+        private ILogger _logger;
+        private CancellationTokenSource _checkLoginRequestStatusCts;
+        private Task _checkLoginRequestStatusTask;
+        private const int REQUEST_TIME_UPDATE_PERIOD_IN_SECONDS = 4;
+        private string _fingerprintPhrase;
+        private string _email;
+        private string _requestId;
+        private string _requestAccessCode;
+        private Tuple<byte[], byte[]> _requestKeyPair;
+        public Action StartTwoFactorAction { get; set; }
+        public Action LogInSuccessAction { get; set; }
+        public Action UpdateTempPasswordAction { get; set; }
+
+        public LoginPasswordlessRequestViewModel()
+        {
+            _deviceActionService = ServiceContainer.Resolve<IDeviceActionService>();
+            _platformUtilsService = ServiceContainer.Resolve<IPlatformUtilsService>();
+            _environmentService = ServiceContainer.Resolve<IEnvironmentService>();
+            _authService = ServiceContainer.Resolve<IAuthService>();
+            _syncService = ServiceContainer.Resolve<ISyncService>();
+            _i18nService = ServiceContainer.Resolve<II18nService>();
+            _stateService = ServiceContainer.Resolve<IStateService>();
+            _cryptoFunctionService = ServiceContainer.Resolve<ICryptoFunctionService>();
+            _cryptoService = ServiceContainer.Resolve<ICryptoService>();
+            _passwordGenerationService = ServiceContainer.Resolve<IPasswordGenerationService>();
+            _logger = ServiceContainer.Resolve<ILogger>("logger");
+
+            PageTitle = AppResources.LogInWithAnotherDevice;
+
+            CreatePasswordlessLoginCommand = new AsyncCommand(async () => await Device.InvokeOnMainThreadAsync(CreatePasswordlessLoginAsync),
+                onException: ex => HandleException(ex),
+                allowsMultipleExecutions: false);
+        }
+
+        public ICommand CreatePasswordlessLoginCommand { get; }
+
+        public string FingerprintPhrase
+        {
+            get => _fingerprintPhrase;
+            set
+            {
+                SetProperty(ref _fingerprintPhrase, value);
+            }
+        }
+
+        public string Email
+        {
+            get => _email;
+            set
+            {
+                SetProperty(ref _email, value);
+            }
+        }
+
+        protected override II18nService i18nService => _i18nService;
+        protected override IEnvironmentService environmentService => _environmentService;
+        protected override IDeviceActionService deviceActionService => _deviceActionService;
+        protected override IPlatformUtilsService platformUtilsService => _platformUtilsService;
```

Comment:
Move this below the other services with an space between the first ones and these ones